lib/Sema/CSSimplify.cpp

```diff
@@ -2701,6 +2701,11 @@ ConstraintSystem::SolutionKind ConstraintSystem::simplifyConformsToConstraint(
     return result;
   }
 
+  // Let's not try to fix missing conformance for Void
+  // and Never because that doesn't really make sense.
+  if (type->isVoid() || type->isUninhabited())
```

Comment:
`Never` can totally conform to things. See [SE-0215](https://github.com/apple/swift-evolution/blob/master/proposals/0215-conform-never-to-hashable-and-equatable.md).

Reply:
Do you think that it would be reasonable for example associated with this PR to diagnose `'foo' requires 'Never' to conform to 'P'`? It just doesn't seem to me that diagnostics like that are really useful...

Reply:
For a certain kind of functional programmer (the kind who uses `Result<Int, Never>` to indicate a value that can't have a failure), that would be useful.

Reply:
Do we know of any cases where this comes up in real-life code?

Reply:
(I guess I do see that in the case you've provided it wouldn't be useful, but I'd want something else there to tell me why it couldn't infer `T`.)

Reply:
Example in this PR is reduced from one of the source compatibility projects @nathawes has been running SourceKit testing on. The problem is that we try `Never` implicitly in multiple places, so it's unclear if benefits of having it "fixed" out-weight in usefulness situations like this where it's just been implicitly attempted because we allow conversions to `Never`.

Reply:
I'm trying to make it diagnose better, not just that `T` can't be inferred, but it's still WIP.